third-party/realdds/src/dds-stream-sensor-bridge.cpp

```diff
@@ -0,0 +1,425 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include <realdds/dds-stream-sensor-bridge.h>
+#include <realdds/dds-device-server.h>
+#include <realdds/dds-stream-server.h>
+#include <realdds/dds-exceptions.h>
+
+#include <rsutils/easylogging/easyloggingpp.h>
+#include <rsutils/json.h>
+
+#include <algorithm>
+#include <iostream>
+
+using nlohmann::json;
+
+
+namespace realdds {
+
+
+dds_stream_sensor_bridge::dds_stream_sensor_bridge() {}
+
+
+dds_stream_sensor_bridge::~dds_stream_sensor_bridge() {}
+
+
+void dds_stream_sensor_bridge::init( std::vector< std::shared_ptr< dds_stream_server > > const & streams )
+{
+    if( streams.empty() )
```

Comment:
We should add this test to `dds_device_server` too. No use of a device without streams.

Reply:
I think we have a test (see `test-device-init.py`) where this is a valid scenario.

Reply:
Then we should reconsider if it is valid. Not a must for this PR, but a point to consider, is there a point in a device without any streams.

Reply:
How about a DFU device? I can see situations where a device is strictly for the control/notification mechanism. I agree it's then not strictly a "device" but we'll only know later if this makes sense or not.